packages/@uppy/companion/src/server/provider/zoom/index.js

```diff
@@ -0,0 +1,262 @@
+          .body
+          .recording_files
+          .find(file => file.id === fileId || file.file_type === fileId)
+
+        if (!file) {
+          return this._downloadError(resp, done)
+        }
+        // timeline files don't have file size, but are typically small json files, should be much less than 1MB
+        const maxExportFileSize = 1024 * 1024
+        done(null, file.file_size || maxExportFileSize)
+      })
+  }
+
+  _initializeData (initialEnd = null) {
+    let end = initialEnd || moment()
+    let start = end.clone().date(1)
+
+    const defaultLimit = end.clone().subtract('months', DEFAULT_RANGE_MOS)
+    const limit = defaultLimit > OLDEST_RECORD ? defaultLimit : OLDEST_RECORD
+
+    const data = {
+      items: []
+    }
+
+    while (start > limit) {
+      start = end.clone().date(1)
+      data.items.push({
+        isFolder: true,
+        icon: 'folder',
+        name: adapter.getDateName(start, end),
+        mimeType: null,
+        id: adapter.getDateFolderId(start, end),
+        thumbnail: null,
+        requestPath: adapter.getDateFolderRequestPath(start, end),
+        modifiedDate: adapter.getDateFolderModified,
+        size: null
+      })
+      end = start.clone().subtract(1, 'days')
+    }
+    data.nextPagePath = adapter.getDateQuery(start)
+    return data
+  }
+
+  _adaptData (results) {
+    if (!results || results.length === 0) {
+      return { items: [] }
+    }
+
+    const data = {
+      nextPagePath: adapter.getQuery(results),
+      items: []
+    }
+    const items = results.meetings || results.recording_files
+    items.forEach(item => {
+      if (item.file_type && item.file_type === 'TIMELINE') {
+        console.log(item.download_url)
+      }
+      data.items.push({
+        isFolder: adapter.getIsFolder(item),
+        icon: adapter.getIcon(item),
+        name: adapter.getItemName(item),
+        mimeType: adapter.getMimeType(item),
+        id: adapter.getId(item),
+        thumbnail: null,
+        requestPath: adapter.getRequestPath(item),
+        modifiedDate: adapter.getStartDate(item),
+        size: null
+      })
+    })
+    return data
+  }
+
+  logout ({ token }, done) {
+    const encodedAuth = Buffer.from(
+      `${process.env.COMPANION_ZOOM_KEY}:${process.env.COMPANION_ZOOM_SECRET}`, 'binary'
+    ).toString('base64')
+
+    return this.client
+      .get('logout')
+      .auth({ basic: encodedAuth })
+      .qs({ token })
+      .request((err, resp) => {
+        if (err || resp.statusCode !== 200) {
+          logger.error(err, 'provider.zoom.logout.error')
+          done(this._error(err, resp))
+          return
+        }
+        done(null, { revoked: true })
+      })
+  }
+
+  _error (err, resp) {
+    const customErrorCodes = [124, 401]
+    if (resp) {
+      const fallbackMessage = `request to ${this.authProvider} returned ${resp.statusCode}`
+      const errMsg = (resp.body && resp.body.error) ? resp.body.error.message : fallbackMessage
```

Comment:
looking at the [API's error definition](https://marketplace.zoom.us/docs/api-reference/error-definitions), It doesn't seem there's an `error` field. Am I missing something?

Reply:
nope, this is just copy and paste error on my part, you're not missing anything with respect to the error field